cli/src/pcluster/resources/imagebuilder/parallelcluster_validate.yaml

```diff
@@ -0,0 +1,507 @@
+              case ${ARCH} in
+                'x86_64')
+                  echo 'x86_64'
+                  ;;
+                'aarch64')
+                  echo 'arm64'
+                  ;;
+                *)
+                  echo "The '${ARCH}' architecture is not supported. Failing build."
+                  exit 1
+                  ;;
+              esac
+
+      # Get platform name
+      - name: PlatformName
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              OS='{{ validate.OperatingSystemName.outputs.stdout }}'
+
+              if [ `echo "${OS}" | grep -E '^(alinux|centos)'` ]; then
+                PLATFORM='RHEL'
+              elif [ `echo "${OS}" | grep -E '^ubuntu'` ]; then
+                PLATFORM='DEBIAN'
+              fi
+
+              echo ${PLATFORM}
+
+      # Get AWS region
+      - name: AWSRegion
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              IMDS_TOKEN=$(curl --retry 3 --retry-delay 0 -s --fail -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 300")
+              AVAIL_ZONE=$(curl --retry 3 --retry-delay 0 -s --fail -H "X-aws-ec2-metadata-token: ${IMDS_TOKEN}" http://169.254.169.254/latest/meta-data/placement/availability-zone)
+              AWS_REGION=${AVAIL_ZONE::-1}
+              echo ${AWS_REGION}
+
+      ### condition ###
+      - name: IntelMPISupported
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              if [ {{ validate.OperatingSystemArchitecture.outputs.stdout }} != 'arm64' ]; then
+                echo "true"
+              else
+                echo "false"
+              fi
+
+      - name: ArmPLSupported
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              if [ {{ validate.OperatingSystemArchitecture.outputs.stdout }} == 'arm64' ]; then
+                echo "true"
+              else
+                echo "false"
+              fi
+
+      - name: LustreSupported
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              ARCHITECTURE='{{ validate.OperatingSystemArchitecture.outputs.stdout }}'
+              OS='{{ validate.OperatingSystemName.outputs.stdout }}'
+              if [ ${ARCHITECTURE} == 'arm64' ] && [[ ${OS} =~ ^(ubuntu(18|20)04|centos8|alinux2)$ ]] || [ ${ARCHITECTURE} == 'x86_64' ]; then
+                echo "true"
+              else
+                echo "false"
+              fi
+
+      ### version ###
+      - name: PythonVersion
+        action: ExecuteBash
+        inputs:
+          commands:
+            - |
+              set -v
+              PATTERN=$(grep -F "default['cluster']['python-version']" /etc/chef/cookbooks/aws-parallelcluster/attributes/default.rb)
```

Comment:
Let's put the `/etc/chef/cookbooks/aws-parallelcluster/attributes/default.rb` 
under a constant, see https://docs.aws.amazon.com/imagebuilder/latest/userguide/image-builder-component-manager-user-defined-variables.html#image-builder-component-manager-user-defined-variables-constants

e.g.
```
constants:
  - CookbookDefaultFile:
      type: string
      value: /etc/chef/cookbooks/aws-parallelcluster/attributes/default.rb
```

Reply:
done